src/main/java/at/ac/tuwien/kr/alpha/common/predicates/External.java

```diff
@@ -0,0 +1,10 @@
+package at.ac.tuwien.kr.alpha.common.predicates;
+
+import at.ac.tuwien.kr.alpha.common.terms.Term;
+import at.ac.tuwien.kr.alpha.grounder.Substitution;
+
+import java.util.List;
+
+public interface External {
+	int evaluate(List<Term> input, List<Term> output, Substitution substitution);
```

Comment:
It is not clear to me why there is `input`, `output` and a `substitution`. On the one hand, the substitution may be used to return variable bindings, on the other hand it maybe is not good to expose the substitution to any custom code (which might try to modify it).

I think it would be best to have only ground input and ground output. By that, the substitution must be handled outside by the grounder, which I guess is the right place anyways, and the `External` is not at all concerned with variables.